state.go

```diff
@@ -309,10 +333,9 @@ func (a *Autopilot) buildServerState(inputs *nextStateInputs, srv raft.Server) S
 
 	var previousHealthy *bool
 
-	a.stateLock.RLock()
```

Comment:
This is what we want to get rid of. The state could have inadvertently changed between when we gathered inputs to build the next state and when we actually are building it. Now we just use the gathered current state on the inputs struct.